docs/current_docs/developer-guide/go/742930-services.mdx

```diff
@@ -0,0 +1,260 @@
+
+This test only runs for the database driver 'pgsql'. Current database driver is 'mysql'.
+/opt/drupal/web/core/tests/Drupal/KernelTests/Core/Database/DriverSpecificKernelTestBase.php:44
+/opt/drupal/vendor/phpunit/phpunit/src/Framework/TestResult.php:728
+
+OK, but incomplete, skipped, or risky tests!
+Tests: 22, Assertions: 72, Skipped: 1.
+```
+
+## Reference: How service binding works in Dagger Functions
+
+If you're not interested in what's happening in the background, you can skip this section and just trust that services are running when they need to be. If you're interested in the theory, keep reading.
+
+Consider this example:
+
+```go file=./snippets/services/service-lifecycle-1/main.go
+```
+
+Here's what happens on the last line:
+
+1. The client requests the `ping` container's stdout, which requires the container to run.
+1. Dagger sees that the `ping` container has a service binding, `redisSrv`.
+1. Dagger starts the `redisSrv` container, which recurses into this same process.
+1. Dagger waits for health checks to pass against `redisSrv`.
+1. Dagger runs the `ping` container with the `redis-srv` alias magically added to `/etc/hosts`.
+
+:::note
+Dagger cancels each service run after a 10 second grace period to avoid frequent restarts.
+:::
+
+Services are based on containers, but they run a little differently. Whereas regular containers in Dagger are de-duplicated across the entire Dagger Engine, service containers are only de-duplicated within a Dagger client session. This means that if you run separate Dagger sessions that use the exact same services, they will each get their own "instance" of the service. This process is carefully tuned to preserve caching at each client call-site, while prohibiting "cross-talk" from one Dagger session's client to another Dagger session's service.
+
+Content-addressed services are very convenient. You don't have to come up with names and maintain instances of services; just use them by value. You also don't have to manage the state of the service; you can just trust that it will be running when needed and stopped when not.
+
+:::tip
+If you need multiple instances of a service, just attach something unique to each one, such as an instance ID.
+:::
+
+Here's a more detailed client-server example of running commands against a Redis service:
+
+```go file=./snippets/services/service-lifecycle-2/main.go
+```
+
+This example relies on the 10-second grace period, which you should try to avoid. Depending on the 10-second grace period is risky because there are many factors which could cause a 10-second delay between calls to Dagger, such as excessive CPU load, high network latency between the client and Dagger, or Dagger operations that require a variable amount of time to process.
+
+It would be better to chain both commands together, which ensures that the service stays running for both.
+
+```go file=./snippets/services/service-lifecycle-3/main.go
+```
```

Comment:
This could probably back-reference to 'explicit start/stop' above as an alternative solution, but it does start to feel a bit redundant